src/routes/tftRoutes.js

```diff
@@ -61,23 +60,29 @@ async function getRecentMatchInfo(req, res) {
         }
 }
 
-async function getAllMatchInfo(req, res) {
+async function getMatchHistoryForSummoner(req, res) {
     const summonerName = req.params.summonerName
-    const count = req.params.count
+    const count = req.query.count
+
     try {
         const summonerInfo = await riotApiClient.fetchTFTSummonerInfo(summonerName);
-        const puuid = summonerInfo.puuid
-        const matchIds = await riotApiClient.getRecentMatches(puuid, count)
-        const matchListInfo = []
-        let i;
-        for(i = 0; i < matchIds.length; i++){
-             matchListInfo.push( await riotApiClient.getMatchData(matchIds[i]))
-             
-        } res.send(matchListInfo)
+        let puuid = summonerInfo.puuid
+        
+        const matches = await riotApiClient.fetchTFTMatchHistory(puuid, count)
+        const summonerAndMatchInfo = {[summonerName] : summonerInfo}
+        summonerAndMatchInfo.matches = matches
+        
+        res.send(summonerAndMatchInfo)
```

Comment:
All of this is close, but not 100% correct. Consider doing it like this:

```suggestion
        const matches = await riotApiClient.fetchTFTMatchHistory(puuid, count)
        const response = { summoner: summonerInfo, matches: matches }

        res.json(response)
```